### PR TITLE
Add Simulate Consume Item

### LINKED ISSUE #544: Simulating Eating
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the feature

Hello!

I think it would be awesome to have the ability to simulate a player eating the item that is in their main hand.

### PR DESCRIPTION
**Pull request by thelooter**
# Description
Adds a way to simulate a Player Consuming an Item. Closes #544

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
- Make sure that unit tests are added which test the relevant changes.
- Make sure that the changes follow the existing code format.

# For maintainer
When a PR is approved, the maintainer should label this PR with `release/*`.

- If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
- If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 1]
```diff
@@ -242,6 +245,34 @@ public boolean reconnect()
 		return true;
 	}
 
+	/**
+	 * Simulates a Player consuming an Edible Item
+	 * @param consumable The Item to consume
+	 */
+	public void simulateConsumeItem(@NotNull ItemStack consumable)
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:262]: This doesn't completely act the same way Paper does. Please look at `net.minecraft.world.entity.LivingEntity#completeUsingItem()` for Paper's implementation.

**Review by Insprill — CHANGES_REQUESTED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java (hunk 1 above)
Comment (thelooter) [reply src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:262]: What do you mean

Comment (Insprill) [reply src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:262]: Sorry, I wrote that a bit hastily.

First some sounds are played at `net.minecraft.world.entity.LivingEntity#completeUsingItem` line 3843

A GenericGameEvent should be fired secondly. `net.minecraft.world.entity.LivingEntity#startUsingItem(InteractionHand, boolean)` line 3766, and `net.minecraft.server.level.ServerLevel#gameEvent` line 1570.

And lastly if the PlayerItemConsumeEvent is cancelled, another GenericGameEvent should be fired. `net.minecraft.world.entity.LivingEntity#completeUsingItem` line 3854, `#stopUsingItem` line 2920, and `net.minecraft.server.level.ServerLevel#gameEvent` line 1570.

**Review by Insprill — APPROVED** [review]
LGTM

**Pull request merged**